src/server/routes/teams/create-team.ts

```diff
@@ -0,0 +1,122 @@
+    || !requestDoc.data.type
+    || requestDoc.data.type !== 'teams'
+    || !requestDoc.data.attributes
+    || !requestDoc.data.attributes.name
+    || typeof requestDoc.data.attributes.name !== 'string') {
+    reply(Boom.badRequest())
+    return
+  }
+
+  const relationships = requestDoc.data.relationships
+  let members: JSONApi.ResourceIdentifierObject[] = []
+  let entries: JSONApi.ResourceIdentifierObject[] = []
+
+  if (relationships) {
+    if (relationships.members && relationships.members.data) {
+      if (!Array.isArray(relationships.members.data)) {
+        reply(Boom.badRequest())
+        return
+      }
+      members = relationships.members.data
+    }
+
+    if (relationships.entries && relationships.entries.data) {
+      if (!Array.isArray(relationships.entries.data)) {
+        reply(Boom.badRequest())
+        return
+      }
+      entries = relationships.entries.data
+    }
+  }
+
+  const team = new TeamModel({
+    teamid: slugify(requestDoc.data.attributes.name),
+    name: requestDoc.data.attributes.name,
+    motto: requestDoc.data.attributes.motto || null,
+    members: [],
+    entries: [],
+  })
+
+  let users: UserModel[] = []
+  let hacks: HackModel[] = []
+
+  if (members.length > 0) {
+    users = await UserModel.find({
+      userid: {
+        $in: members.map((member) => member.id.toString()),
+      },
+    }, '_id userid name').exec()
+    team.members = users.map((user) => user._id)
+  }
+
+  if (entries.length > 0) {
+    hacks = await HackModel.find({
+      hackid: {
+        $in: entries.map((entry) => entry.id.toString()),
+      },
+    }, '_id hackid name').exec()
+    team.entries = hacks.map((hack) => hack._id)
+  }
+
+  try {
+    await team.save()
+  } catch (err) {
+    if (err.code === MongoDBErrors.E11000_DUPLICATE_KEY) {
+      reply(Boom.conflict('Team already exists'))
+      return
+    }
+    throw err
+  }
+
+  const teamResponse: TeamResource.TopLevelDocument = {
+    links: {
+      self: `/teams/${encodeURIComponent(team.teamid)}`,
+    },
+    data: {
+      type: 'teams',
+      id: team.teamid,
+      attributes: {
+        name: team.name,
+        motto: team.motto,
+      },
+      relationships: {
+        members: {
+          links: { self: `/teams/${encodeURIComponent(team.teamid)}/members` },
+          data: users.map((user) => ({ type: 'users', id: user.userid })),
+        },
+        entries: {
+          links: { self: `/teams/${encodeURIComponent(team.teamid)}/entries` },
+          data: hacks.map((hack) => ({ type: 'hacks', id: hack.hackid })),
+        },
+      },
+    },
+  }
+
+  const eventBroadcaster: EventBroadcaster = req.server.app.eventBroadcaster
+  eventBroadcaster.trigger('teams_add', {
+    teamid: team.teamid,
+    name: team.name,
+    motto: team.motto,
+    members: users.map((user) => ({ userid: user.userid, name: user.name })),
+    entries: hacks.map((hack) => ({ hackid: hack.hackid, name: hack.name })),
+  }, req.logger)
```

Comment:
Inside a handler, `pino` has a logger attached to `request.logger` which contains scope for this request. The default logger just logs the line, but the scoped logger outputs request info e.g.:

<pre>
[2017-05-07T09:51:20.428Z] INFO (8088 on DESKTOP-TD7RIL8): Sending Pusher event "users_add" to channel "api_events"
    req: {
      "id": "1494150680422:DESKTOP-TD7RIL8:8088:j2eiq5th:10085",
      "method": "POST",
      "url": "/users",
      "headers": {
        "host": "localhost:12123",
        "accept-encoding": "gzip, deflate",
        "user-agent": "node-superagent/3.5.2",
        "authorization": "Basic dGVzdGF0dGVuZGVlK0xFUEpBSlNLQGhhY2syNC5jby51azpwYXNzd29yZDEyMzQ1Njc4OQ==",
        "content-type": "application/vnd.api+json",
        "content-length": "81",
        "connection": "close"
      },
      "remoteAddress": "127.0.0.1",
      "remotePort": 50993
    }
</pre>

Looks like it's missing in some places so I'll update.